src/EnergyPlus/SolarShading.cc

```diff
@@ -4759,15 +4759,18 @@ void DeterminePolygonOverlap(EnergyPlusData &state,
         if (std::abs(state.dataSolarShading->HCAREA(NS3)) * HCMULT < std::abs(state.dataSolarShading->HCAREA(NS1))) {
             state.dataSolarShading->OverlapStatus = NoOverlap;
         } else {
-            if (state.dataSolarShading->HCAREA(NS1) * state.dataSolarShading->HCAREA(NS2) > 0.0)
+            if (state.dataSolarShading->HCAREA(NS1) * state.dataSolarShading->HCAREA(NS2) > 0.0) {
                 state.dataSolarShading->HCAREA(NS3) = -state.dataSolarShading->HCAREA(NS3); // Determine sign of area of overlap
-            Real64 const HCT_1(state.dataSolarShading->HCT(NS1));
-            Real64 const HCT_2(state.dataSolarShading->HCT(NS2));
-            Real64 HCT_3(HCT_2 * HCT_1); // Determine transmission of overlap
-            if (HCT_2 >= 0.5 && HCT_1 >= 0.5) {
-                if (HCT_2 != 1.0 && HCT_1 != 1.0) {
-                    HCT_3 = 1.0 - HCT_3;
-                }
+            }
+            Real64 const HCT_1 = state.dataSolarShading->HCT(NS1);
+            Real64 const HCT_2 = state.dataSolarShading->HCT(NS2);
+            Real64 HCT_3 = HCT_1 * HCT_2;
+            if (HCT_2 != 1.0 && HCT_1 != 1.0) {
```

Comment:
Isn't this the same as `if (HCT_3 != 1.0)`

Reply:
You could replace line 4767 with `state.dataSolarShading->HCT(NS3) = (HCT_1 + HCT_2) - (HCT_1 * HCT_2);` and delete the `if`.

Reply:
Hmm, at HCT_1 and HCT_2 = 1, `HCT_3 = (HCT_1 * HCT_2) = 1` and `HCT_3 = (HCT_1 + HCT_2) - (HCT_1 * HCT_2); = 1`. And at all other transmittances, `HCT_3 = (HCT_1 + HCT_2) - (HCT_1 * HCT_2);`. Why the need for transmittance checking?

Reply:
I compared this branch to this and there are diffs. I assume it is because either HCT_1 or HCT_2 is 1 but not both where (1 + x) - (1 * x) = 1 and would have been x before. If HCT_1 and HCT_2 are 1 and 0.5 the answer is 0.5. If 0.999999 and 0.5 are used, the answer is 0.9999995. Why is that?

    Real64 const HCT_1 = state.dataSolarShading->HCT(NS1);
    Real64 const HCT_2 = state.dataSolarShading->HCT(NS2);
    state.dataSolarShading->HCT(NS3) = (HCT_1 + HCT_2) - (HCT_1 * HCT_2);